File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/AgentFlowValidationIssue.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * A validation issue for an Agent Flow diagram.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=AgentFlowValidationIssue.Builder.class)

public final class AgentFlowValidationIssue  {
    @Deprecated
    @java.beans.ConstructorProperties({"code", "severity", "phase", "message", "path", "nodeKey", "edgeKey", "resourceKey", "suggestedFix", "details"})
    public AgentFlowValidationIssue(String code, Severity severity, Phase phase, String message, String path, String nodeKey, String edgeKey, String resourceKey, String suggestedFix, java.util.Map<String, Object> details) {
        super();
        this.code = code;
        this.severity = severity;
        this.phase = phase;
        this.message = message;
        this.path = path;
        this.nodeKey = nodeKey;
        this.edgeKey = edgeKey;
        this.resourceKey = resourceKey;
        this.suggestedFix = suggestedFix;
        this.details = details;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Stable machine-readable validation issue code.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("code")
        private String code;

                /**
         * Stable machine-readable validation issue code.
         * @param code the value to set
         * @return this builder
         **/
        

        public Builder code(String code) {
        this.code = code;
        return this;
        }
            /**
     * Issue severity.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("severity")
        private Severity severity;

                /**
         * Issue severity.
         * @param severity the value to set
         * @return this builder
         **/
        

        public Builder severity(Severity severity) {
        this.severity = severity;
        return this;
        }
            /**
     * Validation phase that produced the issue.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("phase")
        private Phase phase;

                /**
         * Validation phase that produced the issue.
         * @param phase the value to set
         * @return this builder
         **/
        

        public Builder phase(Phase phase) {
        this.phase = phase;
        return this;
        }
            /**
     * Human-readable validation issue message.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("message")
        private String message;

                /**
         * Human-readable validation issue message.
         * @param message the value to set
         * @return this builder
         **/
        

        public Builder message(String message) {
        this.message = message;
        return this;
        }
            /**
     * JSON path or logical path to the invalid field.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("path")
        private String path;

                /**
         * JSON path or logical path to the invalid field.
         * @param path the value to set
         * @return this builder
         **/
        

        public Builder path(String path) {
        this.path = path;
        return this;
        }
            /**
     * Optional node key associated with the issue.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("nodeKey")
        private String nodeKey;

                /**
         * Optional node key associated with the issue.
         * @param nodeKey the value to set
         * @return this builder
         **/
        

        public Builder nodeKey(String nodeKey) {
        this.nodeKey = nodeKey;
        return this;
        }
            /**
     * Optional edge key associated with the issue.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("edgeKey")
        private String edgeKey;

                /**
         * Optional edge key associated with the issue.
         * @param edgeKey the value to set
         * @return this builder
         **/
        

        public Builder edgeKey(String edgeKey) {
        this.edgeKey = edgeKey;
        return this;
        }
            /**
     * Optional tool, guardrails, catalog, or workspace key associated with the issue.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("resourceKey")
        private String resourceKey;

                /**
         * Optional tool, guardrails, catalog, or workspace key associated with the issue.
         * @param resourceKey the value to set
         * @return this builder
         **/
        

        public Builder resourceKey(String resourceKey) {
        this.resourceKey = resourceKey;
        return this;
        }
            /**
     * Optional user-facing remediation text.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("suggestedFix")
        private String suggestedFix;

                /**
         * Optional user-facing remediation text.
         * @param suggestedFix the value to set
         * @return this builder
         **/
        

        public Builder suggestedFix(String suggestedFix) {
        this.suggestedFix = suggestedFix;
        return this;
        }
            /**
     * Non-sensitive issue details.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("details")
        private java.util.Map<String, Object> details;

                /**
         * Non-sensitive issue details.
         * @param details the value to set
         * @return this builder
         **/
        

        public Builder details(java.util.Map<String, Object> details) {
        this.details = details;
        return this;
        }


        public AgentFlowValidationIssue build() {
            AgentFlowValidationIssue model = new AgentFlowValidationIssue(this.code
                    , this.severity
                    , this.phase
                    , this.message
                    , this.path
                    , this.nodeKey
                    , this.edgeKey
                    , this.resourceKey
                    , this.suggestedFix
                    , this.details);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(AgentFlowValidationIssue model) {
                this.code(model.getCode());
    this.severity(model.getSeverity());
    this.phase(model.getPhase());
    this.message(model.getMessage());
    this.path(model.getPath());
    this.nodeKey(model.getNodeKey());
    this.edgeKey(model.getEdgeKey());
    this.resourceKey(model.getResourceKey());
    this.suggestedFix(model.getSuggestedFix());
    this.details(model.getDetails());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Stable machine-readable validation issue code.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("code")
    private final String code;

        /**
     * Stable machine-readable validation issue code.
     * @return the value
     **/
    
    public String getCode() {
        return code;
    }

    /**
     * Issue severity.
     **/
    public enum Severity {
        Error("ERROR"),
        Warning("WARNING"),
        Info("INFO"),
        ;

        

        private final String value;
        private static java.util.Map<String, Severity> map;

        static {
            map = new java.util.HashMap<>();
            for (Severity v : Severity.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        Severity(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static Severity create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid Severity: " + key);
        }
    };
        /**
     * Issue severity.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("severity")
    private final Severity severity;

        /**
     * Issue severity.
     * @return the value
     **/
    
    public Severity getSeverity() {
        return severity;
    }

    /**
     * Validation phase that produced the issue.
     **/
    public enum Phase {
        Model("MODEL"),
        Graph("GRAPH"),
        Config("CONFIG"),
        Reference("REFERENCE"),
        Codegen("CODEGEN"),
        ;

        

        private final String value;
        private static java.util.Map<String, Phase> map;

        static {
            map = new java.util.HashMap<>();
            for (Phase v : Phase.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        Phase(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static Phase create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid Phase: " + key);
        }
    };
        /**
     * Validation phase that produced the issue.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("phase")
    private final Phase phase;

        /**
     * Validation phase that produced the issue.
     * @return the value
     **/
    
    public Phase getPhase() {
        return phase;
    }


        /**
     * Human-readable validation issue message.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("message")
    private final String message;

        /**
     * Human-readable validation issue message.
     * @return the value
     **/
    
    public String getMessage() {
        return message;
    }


        /**
     * JSON path or logical path to the invalid field.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("path")
    private final String path;

        /**
     * JSON path or logical path to the invalid field.
     * @return the value
     **/
    
    public String getPath() {
        return path;
    }


        /**
     * Optional node key associated with the issue.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("nodeKey")
    private final String nodeKey;

        /**
     * Optional node key associated with the issue.
     * @return the value
     **/
    
    public String getNodeKey() {
        return nodeKey;
    }


        /**
     * Optional edge key associated with the issue.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("edgeKey")
    private final String edgeKey;

        /**
     * Optional edge key associated with the issue.
     * @return the value
     **/
    
    public String getEdgeKey() {
        return edgeKey;
    }


        /**
     * Optional tool, guardrails, catalog, or workspace key associated with the issue.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("resourceKey")
    private final String resourceKey;

        /**
     * Optional tool, guardrails, catalog, or workspace key associated with the issue.
     * @return the value
     **/
    
    public String getResourceKey() {
        return resourceKey;
    }


        /**
     * Optional user-facing remediation text.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("suggestedFix")
    private final String suggestedFix;

        /**
     * Optional user-facing remediation text.
     * @return the value
     **/
    
    public String getSuggestedFix() {
        return suggestedFix;
    }


        /**
     * Non-sensitive issue details.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("details")
    private final java.util.Map<String, Object> details;

        /**
     * Non-sensitive issue details.
     * @return the value
     **/
    
    public java.util.Map<String, Object> getDetails() {
        return details;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("AgentFlowValidationIssue(");
        sb.append("code=").append(String.valueOf(this.code));
        sb.append(", severity=").append(String.valueOf(this.severity));
        sb.append(", phase=").append(String.valueOf(this.phase));
        sb.append(", message=").append(String.valueOf(this.message));
        sb.append(", path=").append(String.valueOf(this.path));
        sb.append(", nodeKey=").append(String.valueOf(this.nodeKey));
        sb.append(", edgeKey=").append(String.valueOf(this.edgeKey));
        sb.append(", resourceKey=").append(String.valueOf(this.resourceKey));
        sb.append(", suggestedFix=").append(String.valueOf(this.suggestedFix));
        sb.append(", details=").append(String.valueOf(this.details));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AgentFlowValidationIssue)) {
            return false;
        }

        AgentFlowValidationIssue other = (AgentFlowValidationIssue) o;
        return java.util.Objects.equals(this.code, other.code) &&
            java.util.Objects.equals(this.severity, other.severity) &&
            java.util.Objects.equals(this.phase, other.phase) &&
            java.util.Objects.equals(this.message, other.message) &&
            java.util.Objects.equals(this.path, other.path) &&
            java.util.Objects.equals(this.nodeKey, other.nodeKey) &&
            java.util.Objects.equals(this.edgeKey, other.edgeKey) &&
            java.util.Objects.equals(this.resourceKey, other.resourceKey) &&
            java.util.Objects.equals(this.suggestedFix, other.suggestedFix) &&
            java.util.Objects.equals(this.details, other.details);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.code == null ? 43 : this.code.hashCode());
        result = (result * PRIME) + (this.severity == null ? 43 : this.severity.hashCode());
        result = (result * PRIME) + (this.phase == null ? 43 : this.phase.hashCode());
        result = (result * PRIME) + (this.message == null ? 43 : this.message.hashCode());
        result = (result * PRIME) + (this.path == null ? 43 : this.path.hashCode());
        result = (result * PRIME) + (this.nodeKey == null ? 43 : this.nodeKey.hashCode());
        result = (result * PRIME) + (this.edgeKey == null ? 43 : this.edgeKey.hashCode());
        result = (result * PRIME) + (this.resourceKey == null ? 43 : this.resourceKey.hashCode());
        result = (result * PRIME) + (this.suggestedFix == null ? 43 : this.suggestedFix.hashCode());
        result = (result * PRIME) + (this.details == null ? 43 : this.details.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/agent_flow_validation_issue.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class AgentFlowValidationIssue(object):
    """
    A validation issue for an Agent Flow diagram.
    """

    #: A constant which can be used with the severity property of a AgentFlowValidationIssue.
    #: This constant has a value of "ERROR"
    SEVERITY_ERROR = "ERROR"

    #: A constant which can be used with the severity property of a AgentFlowValidationIssue.
    #: This constant has a value of "WARNING"
    SEVERITY_WARNING = "WARNING"

    #: A constant which can be used with the severity property of a AgentFlowValidationIssue.
    #: This constant has a value of "INFO"
    SEVERITY_INFO = "INFO"

    #: A constant which can be used with the phase property of a AgentFlowValidationIssue.
    #: This constant has a value of "MODEL"
    PHASE_MODEL = "MODEL"

    #: A constant which can be used with the phase property of a AgentFlowValidationIssue.
    #: This constant has a value of "GRAPH"
    PHASE_GRAPH = "GRAPH"

    #: A constant which can be used with the phase property of a AgentFlowValidationIssue.
    #: This constant has a value of "CONFIG"
    PHASE_CONFIG = "CONFIG"

    #: A constant which can be used with the phase property of a AgentFlowValidationIssue.
    #: This constant has a value of "REFERENCE"
    PHASE_REFERENCE = "REFERENCE"

    #: A constant which can be used with the phase property of a AgentFlowValidationIssue.
    #: This constant has a value of "CODEGEN"
    PHASE_CODEGEN = "CODEGEN"

    def __init__(self, **kwargs):
        """
        Initializes a new AgentFlowValidationIssue object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param code:
            The value to assign to the code property of this AgentFlowValidationIssue.
        :type code: str

        :param severity:
            The value to assign to the severity property of this AgentFlowValidationIssue.
            Allowed values for this property are: "ERROR", "WARNING", "INFO"
        :type severity: str

        :param phase:
            The value to assign to the phase property of this AgentFlowValidationIssue.
            Allowed values for this property are: "MODEL", "GRAPH", "CONFIG", "REFERENCE", "CODEGEN"
        :type phase: str

        :param message:
            The value to assign to the message property of this AgentFlowValidationIssue.
        :type message: str

        :param path:
            The value to assign to the path property of this AgentFlowValidationIssue.
        :type path: str

        :param node_key:
            The value to assign to the node_key property of this AgentFlowValidationIssue.
        :type node_key: str

        :param edge_key:
            The value to assign to the edge_key property of this AgentFlowValidationIssue.
        :type edge_key: str

        :param resource_key:
            The value to assign to the resource_key property of this AgentFlowValidationIssue.
        :type resource_key: str

        :param suggested_fix:
            The value to assign to the suggested_fix property of this AgentFlowValidationIssue.
        :type suggested_fix: str

        :param details:
            The value to assign to the details property of this AgentFlowValidationIssue.
        :type details: dict(str, object)

        """
        self.swagger_types = {
            'code': 'str',
            'severity': 'str',
            'phase': 'str',
            'message': 'str',
            'path': 'str',
            'node_key': 'str',
            'edge_key': 'str',
            'resource_key': 'str',
            'suggested_fix': 'str',
            'details': 'dict(str, object)'
        }

        self.attribute_map = {
            'code': 'code',
            'severity': 'severity',
            'phase': 'phase',
            'message': 'message',
            'path': 'path',
            'node_key': 'nodeKey',
            'edge_key': 'edgeKey',
            'resource_key': 'resourceKey',
            'suggested_fix': 'suggestedFix',
            'details': 'details'
        }

        self._code = None
        self._severity = None
        self._phase = None
        self._message = None
        self._path = None
        self._node_key = None
        self._edge_key = None
        self._resource_key = None
        self._suggested_fix = None
        self._details = None

    @property
    def code(self):
        """
        **[Required]** Gets the code of this AgentFlowValidationIssue.
        Stable machine-readable validation issue code.


        :return: The code of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._code

    @code.setter
    def code(self, code):
        """
        Sets the code of this AgentFlowValidationIssue.
        Stable machine-readable validation issue code.


        :param code: The code of this AgentFlowValidationIssue.
        :type: str
        """
        self._code = code

    @property
    def severity(self):
        """
        **[Required]** Gets the severity of this AgentFlowValidationIssue.
        Issue severity.

        Allowed values for this property are: "ERROR", "WARNING", "INFO"


        :return: The severity of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._severity

    @severity.setter
    def severity(self, severity):
        """
        Sets the severity of this AgentFlowValidationIssue.
        Issue severity.


        :param severity: The severity of this AgentFlowValidationIssue.
        :type: str
        """
        allowed_values = ["ERROR", "WARNING", "INFO"]
        if not value_allowed_none_or_none_sentinel(severity, allowed_values):
            raise ValueError(
                "Invalid value for `severity`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._severity = severity

    @property
    def phase(self):
        """
        **[Required]** Gets the phase of this AgentFlowValidationIssue.
        Validation phase that produced the issue.

        Allowed values for this property are: "MODEL", "GRAPH", "CONFIG", "REFERENCE", "CODEGEN"


        :return: The phase of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._phase

    @phase.setter
    def phase(self, phase):
        """
        Sets the phase of this AgentFlowValidationIssue.
        Validation phase that produced the issue.


        :param phase: The phase of this AgentFlowValidationIssue.
        :type: str
        """
        allowed_values = ["MODEL", "GRAPH", "CONFIG", "REFERENCE", "CODEGEN"]
        if not value_allowed_none_or_none_sentinel(phase, allowed_values):
            raise ValueError(
                "Invalid value for `phase`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._phase = phase

    @property
    def message(self):
        """
        **[Required]** Gets the message of this AgentFlowValidationIssue.
        Human-readable validation issue message.


        :return: The message of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._message

    @message.setter
    def message(self, message):
        """
        Sets the message of this AgentFlowValidationIssue.
        Human-readable validation issue message.


        :param message: The message of this AgentFlowValidationIssue.
        :type: str
        """
        self._message = message

    @property
    def path(self):
        """
        Gets the path of this AgentFlowValidationIssue.
        JSON path or logical path to the invalid field.


        :return: The path of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._path

    @path.setter
    def path(self, path):
        """
        Sets the path of this AgentFlowValidationIssue.
        JSON path or logical path to the invalid field.


        :param path: The path of this AgentFlowValidationIssue.
        :type: str
        """
        self._path = path

    @property
    def node_key(self):
        """
        Gets the node_key of this AgentFlowValidationIssue.
        Optional node key associated with the issue.


        :return: The node_key of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._node_key

    @node_key.setter
    def node_key(self, node_key):
        """
        Sets the node_key of this AgentFlowValidationIssue.
        Optional node key associated with the issue.


        :param node_key: The node_key of this AgentFlowValidationIssue.
        :type: str
        """
        self._node_key = node_key

    @property
    def edge_key(self):
        """
        Gets the edge_key of this AgentFlowValidationIssue.
        Optional edge key associated with the issue.


        :return: The edge_key of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._edge_key

    @edge_key.setter
    def edge_key(self, edge_key):
        """
        Sets the edge_key of this AgentFlowValidationIssue.
        Optional edge key associated with the issue.


        :param edge_key: The edge_key of this AgentFlowValidationIssue.
        :type: str
        """
        self._edge_key = edge_key

    @property
    def resource_key(self):
        """
        Gets the resource_key of this AgentFlowValidationIssue.
        Optional tool, guardrails, catalog, or workspace key associated with the issue.


        :return: The resource_key of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._resource_key

    @resource_key.setter
    def resource_key(self, resource_key):
        """
        Sets the resource_key of this AgentFlowValidationIssue.
        Optional tool, guardrails, catalog, or workspace key associated with the issue.


        :param resource_key: The resource_key of this AgentFlowValidationIssue.
        :type: str
        """
        self._resource_key = resource_key

    @property
    def suggested_fix(self):
        """
        Gets the suggested_fix of this AgentFlowValidationIssue.
        Optional user-facing remediation text.


        :return: The suggested_fix of this AgentFlowValidationIssue.
        :rtype: str
        """
        return self._suggested_fix

    @suggested_fix.setter
    def suggested_fix(self, suggested_fix):
        """
        Sets the suggested_fix of this AgentFlowValidationIssue.
        Optional user-facing remediation text.


        :param suggested_fix: The suggested_fix of this AgentFlowValidationIssue.
        :type: str
        """
        self._suggested_fix = suggested_fix

    @property
    def details(self):
        """
        Gets the details of this AgentFlowValidationIssue.
        Non-sensitive issue details.


        :return: The details of this AgentFlowValidationIssue.
        :rtype: dict(str, object)
        """
        return self._details

    @details.setter
    def details(self, details):
        """
        Sets the details of this AgentFlowValidationIssue.
        Non-sensitive issue details.


        :param details: The details of this AgentFlowValidationIssue.
        :type: dict(str, object)
        """
        self._details = details

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
